addons/web/controllers/export.py

```diff
@@ -328,42 +328,102 @@ def get_fields(self, model, prefix='', parent_name='',
+        for fname, field in exportable_fields.items():
+            if field.get('type') != 'properties':
+                continue
+
+            definition_record = field['definition_record']
+            definition_record_field = field['definition_record_field']
+
+            target_model = Model.env[Model._fields[definition_record].comodel_name]
+            # Depends of the records selected to avoid showing useless Properties
+            self_subquery = self.active_test(False)._search(domain)
+            definition_records = target_model.search_fetch(
+                [
+                    ('id', 'in', self_subquery.subselect(definition_record)),
+                    (definition_record_field, '!=', False),
+                ],
+                [definition_record_field, 'display_name'],
+                order='id',  # Avoid complex order
+            )
+
+            for record in definition_records:
+                for definition in record[definition_record_field]:
+                    # definition = {
+                    #     'name': 'aa34746a6851ee4e',
+                    #     'string': 'Partner',
+                    #     'type': 'many2one',
+                    #     'comodel': 'test_new_api.partner',
+                    #     'default': [1337, 'Bob'],
+                    # }
+                    if (
+                        definition['type'] == 'separator' or
+                        (
+                            definition['type'] in ('many2one', 'many2many')
+                            and definition['comodel'] not in Model.env
+                        )
+                    ):
+                        continue
+                    id_field = f"{fname}.{definition['name']}"
+                    property_fields[id_field] = {
+                        'type': definition['type'],
+                        'string': Model.env._(
+                            "%(property_string)s (%(parent_name)s)",
+                            property_string=definition['string'], parent_name=record.display_name,
+                        ),
+                        'default_export_compatible': field['default_export_compatible'],
+                    }
+                    if definition['type'] in ('many2one', 'many2many'):
+                        property_fields[id_field]['relation'] = definition['comodel']
+
+        exportable_fields.update(property_fields)
+
+        fields_sequence = sorted(exportable_fields.items(), key=lambda field: field[1]['string'].lower())
+
+        result = []
+        for field_name, field in fields_sequence:
             ident = prefix + ('/' if prefix else '') + field_name
             val = ident
             if field_name == 'name' and import_compat and parent_field_type in ['many2one', 'many2many']:
                 # Add name field when expand m2o and m2m fields in import-compatible mode
                 val = prefix
             name = parent_name + (parent_name and '/' or '') + field['string']
```

Comment:
Don't mix and/or.
```suggestion
            name = parent_name + ('/' if parent_name else '') + field['string']
```

Reply:
I won't refactor all the code nearby in this PR. A proper refactor will be done in 19.0 (I hope) :wink: